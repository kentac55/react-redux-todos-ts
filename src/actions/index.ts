import actionCreatorFactory from 'typescript-fsa'
import { AsyncOpResult, Todo, VisibilityFilter } from '../types'

const actionCreator = actionCreatorFactory()

type AddTodoProp = {
  text: string
}
export const addTodo = actionCreator.async<
  AddTodoProp,
  AsyncOpResult<Todo>,
  Error
>('AddTodo')

type ToggleTodoProp = {
  todo: Todo
}
export const toggleTodo = actionCreator.async<
  ToggleTodoProp,
  AsyncOpResult<Todo>,
  Error
>('ToggleTodo')

export const setVisibilityFilter = actionCreator<VisibilityFilter>(
  'SetVisibilityFilter'
)

type GetTodoProp = {
  id: number
}
export const getTodo = actionCreator.async<
  GetTodoProp,
  AsyncOpResult<Todo>,
  Error
>('GetTodo')

type GetTodosProp = void
export const getTodos = actionCreator.async<
  GetTodosProp,
  AsyncOpResult<Todo[]>,
  Error
>('GetTogos')

type InitProp = void
export const initApp = actionCreator.async<
  InitProp,
  AsyncOpResult<Todo[]>,
  Error
>('InitApp')
