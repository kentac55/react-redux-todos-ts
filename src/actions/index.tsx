import actionCreatorFactory from 'typescript-fsa'
import { VisibilityFilterKinds } from '../types'

const actionCreator = actionCreatorFactory()
export const addTodoAction = actionCreator<string>('ADD')
export const toggleTodoAction = actionCreator<number>('TOGGLE')

export const setVisibilityFilterAction = actionCreator<VisibilityFilterKinds>(
  'SET'
)
