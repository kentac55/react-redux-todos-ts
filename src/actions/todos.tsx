let nextTodoId = 0
export const addTodoAction = (text: string) =>
  ({
    type: 'ADD',
    id: nextTodoId++,
    text,
  } as const)

export const toggleTodoAction = (id: number) =>
  ({
    type: 'TOGGLE',
    id,
  } as const)
