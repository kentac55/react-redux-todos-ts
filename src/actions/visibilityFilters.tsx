import { VisibilityFilterKinds } from '../types'

export const setVisibilityFilterAction = (filter: VisibilityFilterKinds) =>
  ({
    type: 'SET_VISIBILITY_FILTER',
    filter,
  } as const)
