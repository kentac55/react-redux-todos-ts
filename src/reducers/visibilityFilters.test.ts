import { Action } from 'redux'
import {
  visibilityFilterReducer,
  VisibilityFilterState,
} from './visibilityFilters'
import { setVisibilityFilter } from '../actions'

const all = {
  visibilityFilter: 'SHOW_ALL',
} as const
const completed = {
  visibilityFilter: 'SHOW_COMPLETED',
} as const
const active = {
  visibilityFilter: 'SHOW_ACTIVE',
} as const
const init = all

it('should return initial state', () => {
  expect(visibilityFilterReducer(undefined, { type: undefined })).toEqual(init)
  expect(
    visibilityFilterReducer(undefined, { type: undefined })
  ).toMatchSnapshot()
})

it.each<[Action, VisibilityFilterState]>([
  [setVisibilityFilter('SHOW_ALL'), all],
  [setVisibilityFilter('SHOW_COMPLETED'), completed],
  [setVisibilityFilter('SHOW_ACTIVE'), active],
])('should handle %s', (action, expectedState) => {
  expect(visibilityFilterReducer(init, action)).toStrictEqual(expectedState)
  expect(visibilityFilterReducer(init, action)).toMatchSnapshot()
})
