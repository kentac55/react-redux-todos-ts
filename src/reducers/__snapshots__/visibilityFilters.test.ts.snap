// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle { type: 'SetVisibilityFilter', payload: 'SHOW_ACTIVE' } 1`] = `
Object {
  "visibilityFilter": "SHOW_ACTIVE",
}
`;

exports[`should handle { type: 'SetVisibilityFilter', payload: 'SHOW_ALL' } 1`] = `
Object {
  "visibilityFilter": "SHOW_ALL",
}
`;

exports[`should handle { type: 'SetVisibilityFilter', payload: 'SHOW_COMPLETED' } 1`] = `
Object {
  "visibilityFilter": "SHOW_COMPLETED",
}
`;

exports[`should return initial state 1`] = `
Object {
  "visibilityFilter": "SHOW_ALL",
}
`;
